rflx/pyrflx/typevalue.py

```diff
@@ -1077,24 +1099,26 @@ def __update_simplified_mapping(self, field: Optional[Field] = None) -> None:
             self._simplified_mapping[field.name_length] = field.typeval.size
             self._simplified_mapping[field.name_first] = field.first
             self._simplified_mapping[field.name_last] = field.last
+            self._simplified_mapping[self.message_last_name] = field.last
             return
 
         self._simplified_mapping = {}
         for v in self._fields.values():
-            if not v.set:
-                continue
-            if isinstance(v.typeval, ScalarValue):
+            if isinstance(v.typeval, ScalarValue) and v.set:
                 self._simplified_mapping[v.name_variable] = v.typeval.expr
-            self._simplified_mapping[v.name_length] = v.typeval.size
-            self._simplified_mapping[v.name_first] = v.first
-            self._simplified_mapping[v.name_last] = v.last
+            if isinstance(v.typeval, ScalarValue) or v.set:
+                self._simplified_mapping[v.name_length] = v.typeval.size
+            if isinstance(v.first, Number):
+                self._simplified_mapping[v.name_first] = v.first
+            if isinstance(v.last, Number):
+                self._simplified_mapping[v.name_last] = v.last
```

Comment:
`isinstance` checks seem to be unnecessary.

Reply:
We only want to add these values to the mapping if they're known.

Reply:
Then unit tests should be added in which these checks make a difference. Currently, no test fails if the checks are removed.

Reply:
Functionally it doesn't make a difference but it is a slight performance optimization that prevents unnecessary expression evaluations.